lib/perl/Genome/Model/Tools/Picard/CompareSams.pm

```diff
@@ -39,10 +39,10 @@ sub _jar_name {
     return 'CompareSAMs.jar';
 }
 
-sub _java_class_name {
+sub _java_class {
     my $self = shift;
-    return 'net.sf.samtools.apps.CompareSAMs' if $self->use_version eq '1.17';
-    return 'net.sf.picard.sam.CompareSAMs';
+    return qw(samtools apps CompareSAMs) if $self->use_version eq '1.17';
```

Comment:
I know this isn't related directly to your change, but is it really only version 1.17 where the name is different? Not all versions prior to (or after) 1.17?